lib/ansible/module_utils/facts/system/ssh_pub_keys.py

```diff
@@ -47,6 +47,10 @@ def collect(self, module=None, collected_facts=None):
                 key_filename = '%s/ssh_host_%s_key.pub' % (keydir, type_)
                 keydata = get_file_content(key_filename)
                 if keydata is not None:
-                    ssh_pub_key_facts[factname] = keydata.split()[1]
+                    (subtype, key) = keydata.split()
+                    ssh_pub_key_facts[factname] = key
+                    if type_ == 'ecdsa':
```

Comment:
I agree. How about we just call it "keytype" instead of "subtype" and add it for all of them without blacklists? It's a bit redundant, but should be harmless and make future work easier both for facts maintenance and for other code that will use these facts.

I lean toward "keytype" as the term used in the `man` page for `sshd` on my Fedora 25 system:
```
The keytype is “ecdsa-sha2-nistp256”, “ecdsa-sha2-nistp384”, “ecdsa-sha2-nistp521”, “ssh-ed25519”, “ssh-dss” or “ssh-rsa”...
```